homeassistant/components/vorwerk/vacuum.py

```diff
@@ -0,0 +1,348 @@
+
+        try:
+            self._state = self.robot.state
+        except NeatoRobotException as ex:
+            if self._available:  # print only once when available
+                _LOGGER.error(
+                    "Vorwerk vacuum connection error for '%s': %s", self.entity_id, ex
+                )
+            self._state = None
+            self._available = False
+            return
+
+        self._available = True
+        _LOGGER.debug("self._state=%s", self._state)
+        if "alert" in self._state:
+            robot_alert = ALERTS.get(self._state["alert"])
+        else:
+            robot_alert = None
+        if self._state["state"] == 1:
+            if self._state["details"]["isCharging"]:
+                self._clean_state = STATE_DOCKED
+                self._status_state = "Charging"
+            elif (
+                self._state["details"]["isDocked"]
+                and not self._state["details"]["isCharging"]
+            ):
+                self._clean_state = STATE_DOCKED
+                self._status_state = "Docked"
+            else:
+                self._clean_state = STATE_IDLE
+                self._status_state = "Stopped"
+
+            if robot_alert is not None:
+                self._status_state = robot_alert
+        elif self._state["state"] == 2:
+            if robot_alert is None:
+                self._clean_state = STATE_CLEANING
+                self._status_state = (
+                    f"{MODE.get(self._state['cleaning']['mode'])} "
+                    f"{ACTION.get(self._state['action'])}"
+                )
+                if (
+                    "boundary" in self._state["cleaning"]
+                    and "name" in self._state["cleaning"]["boundary"]
+                ):
+                    self._status_state += (
+                        f" {self._state['cleaning']['boundary']['name']}"
+                    )
+            else:
+                self._status_state = robot_alert
+        elif self._state["state"] == 3:
+            self._clean_state = STATE_PAUSED
+            self._status_state = "Paused"
+        elif self._state["state"] == 4:
+            self._clean_state = STATE_ERROR
+            self._status_state = ERRORS.get(self._state["error"])
+
+        self._battery_level = self._state["details"]["charge"]
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_VORWERK
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return self._battery_level
+
+    @property
+    def available(self):
+        """Return if the robot is available."""
+        return self._available
+
+    @property
+    def icon(self):
+        """Return specific icon."""
+        return "mdi:robot-vacuum-variant"
```

Comment:
Not dynamic based on state? docked vs moving vs error? (not sure if there are mdi's for that, but would be fun!)

Reply:
Good idea! I will add this feature in a later version.

Reply:
Good Idee. I will add this in a later version.

Reply:
Currently there is only that static icon available.